torch/lib/c10d/HashStore.cpp

```diff
@@ -0,0 +1,84 @@
+#include <c10d/HashStore.hpp>
+
+#include <stdint.h>
+#include <unistd.h>
+
+#include <errno.h>
+#include <chrono>
+#include <cstdio>
+#include <system_error>
+
+namespace c10d {
+
+void HashStore::set(const std::string& key, const std::vector<uint8_t>& data) {
+  std::unique_lock<std::mutex> lock(m_);
+  map_[key] = data;
+  cv_.notify_all();
+}
+
+std::vector<uint8_t> HashStore::get(const std::string& key) {
+  std::unique_lock<std::mutex> lock(m_);
+  auto it = map_.find(key);
+  if (it != map_.end()) {
+    return it->second;
+  }
+  // Not found? Wait up to any Store timeout_.
+  lock.unlock();
+  wait({key}, timeout_);
+  lock.lock();
+  return map_[key];
```

Comment:
`cv_.wait_for`? The `pred` can be simpler than the one used in `wait`.

Reply:
sure, done.